### ISSUE TITLE
Residual heat from DWH cycle still counted as DWH for efficiency

### ISSUE THREAD

**Issue by MrTimbones** [MyForest/mmspheatpump#19]
After heating the hot water this morning (until 10:37), there was a sizeable gap between the end of that and the start of the space heating (11:10), yet the Efficiency chart still counts the new heat generated as being with the hot water. I would have expected those last two blue bars to be green...

![image](https://user-images.githubusercontent.com/75039652/157432745-7f89456f-0ce5-4d4b-af8c-f2a37095275a.png)

**Comment by MrTimbones** [comment]
I've figured out this that is an artefact of how MELCloud reports energy consumption, attributing it to whichever of "Heating" or "HotWater" was running last. This means that energy consumed while the HP is idle (10:39 - 11:10) is still counted as "HotWater". It then takes a minute or two to switch over to "Heating" when that starts, and some of that new energy (at 11:10) is still incorrectly counted as "HotWater".

In other words, the _HeatingEnergyConsumedRate1_ and _HotWaterEnergyConsumedRate1_ (and *_ProducedRate1_) fields from MELCloud are unreliable. This won't have been an issue for anyone who wasn't using these fields.

I'm now correcting this when collecting the data by matching the Heating vs. HotWater consumption/production to the status of the pumps, and recording "Idle" consumption in a separate feed.

**Comment by MyForest** [comment]
Yep @MrTimbones I've come to accept it as somewhat of a good thing because in my setup the cost of the system idling is attributed to the last thing that was running. As a result, in the Summer the DHW gets dinged for the cost of keeping the monitoring active!
